pkg/gohai/memory/memory_windows.go

```diff
@@ -26,28 +28,19 @@ type MEMORYSTATUSEX struct {
 	ulAvailExtendedVirtual uint64 // reserved (always zero)
 }
 
-func getMemoryInfo() (memoryInfo map[string]string, err error) {
-	memoryInfo = make(map[string]string)
-
-	mem, _, _, err := getMemoryInfoByte()
-	if err == nil {
-		memoryInfo["total"] = strconv.FormatUint(mem, 10)
-	}
-	return
-}
-
-func getMemoryInfoByte() (mem uint64, swap uint64, warnings []string, err error) {
+func (info *Info) fillMemoryInfo() {
 	var mod = syscall.NewLazyDLL("kernel32.dll")
 	var getMem = mod.NewProc("GlobalMemoryStatusEx")
 
 	var memStruct MEMORYSTATUSEX
-
 	memStruct.dwLength = uint32(unsafe.Sizeof(memStruct))
 
 	status, _, err := getMem.Call(uintptr(unsafe.Pointer(&memStruct)))
 	if status != 0 {
-		mem = memStruct.ulTotalPhys
-		err = nil
+		info.TotalBytes = utils.NewValue(memStruct.ulTotalPhys)
+	} else {
+		info.TotalBytes = utils.NewErrorValue[uint64](err)
 	}
-	return
+
+	info.SwapTotalKb = utils.NewErrorValue[uint64](fmt.Errorf("memory.SwapTotalKb: %w", utils.ErrNotCollectable))
```

Comment:
Before the value seems to not be collected. Do you still have the same behaviour as before?

Reply:
Note that this is an error (saying that the field cannot be collected on that platform), not a value.

If the `Value[T]` is an error then it's not included in the json which is produced, so the behavior is kept.